sync-core/src/main/java/com/cloudant/sync/datastore/DocumentRevisionBuilder.java

```diff
@@ -211,6 +222,21 @@ public MutableDocumentRevision buildMutable() {
     }
 
     /**
+     * Builds and returns the {@link com.cloudant.sync.datastore.ProjectedDocumentRevision} for this builder.
+     * @return {@link com.cloudant.sync.datastore.ProjectedDocumentRevision} for this builder.
+     */
+    public ProjectedDocumentRevision buildProjected() {
+        BasicDocumentRevision.BasicDocumentRevisionOptions options = new BasicDocumentRevision.BasicDocumentRevisionOptions();
+        options.sequence = sequence;
+        options.docInternalId = docInternalId;
+        options.deleted = deleted;
+        options.current = current;
+        options.parent = parent;
+        options.attachments = attachments;
+        return new ProjectedDocumentRevision(docId, revId, body, options, datastore);
+    }
```

Comment:
I see there isn't an easy way to do something that isn't public given our class structure. Let's leave this for now.